### PR TITLE
Fix ELWCExampleGen

### PR DESCRIPTION
**Pull request by casassg**
Due to how BaseExampleGen writes splits, ELWCExampleGen needs to serialize the data before sending it to the super class. This was causing an assertion error, as Beam was trying to map the protobuf class to a byte writer from BaseExampleGen.

Collab showing fix: https://colab.research.google.com/drive/1KHEG7Yv1_JLB132ZnKLOxIANXwWaO6X3?usp=sharing

Fixes: #3610

### CODE REVIEW

**Review by 1025KB — APPROVED** [review]

**Comment by casassg** [comment]
@1025KB ready to be merged whenever :D Ran test successfully 
```
python tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor_test.py
```

Had to deserialize on beam, as otherwise bytes may not be used for the assertion (not 100% sure why)

**Comment by casassg** [comment]
@1025KB any chance to merge this before next release?

**Review by 1025KB — APPROVED** [review]

**Comment by casassg** [comment]
@1025KB the error getting atm seems to be due to test running agains latest MLMD 

```
  File "/home/runner/work/tfx/tfx/tfx/types/system_artifacts.py", line 40, in Dataset
    MLMD_SYSTEM_BASE_TYPE = mlmd_types.Dataset().system_type
AttributeError: 'Dataset' object has no attribute 'system_type'
```

Seems this is due to https://github.com/google/ml-metadata/commit/aa492c474f8b3a54dbb6461ddb7b35c7e69b4a47 not being used to test on the tfx-ci-test in GitHub (guessing internal tests should pass as it uses source files)

**Comment by 1025KB** [comment]
> @1025KB the error getting atm seems to be due to test running agains latest MLMD
> 
> ```
>   File "/home/runner/work/tfx/tfx/tfx/types/system_artifacts.py", line 40, in Dataset
>     MLMD_SYSTEM_BASE_TYPE = mlmd_types.Dataset().system_type
> AttributeError: 'Dataset' object has no attribute 'system_type'
> ```
> 
> Seems this is due to [google/ml-metadata@aa492c4](https://github.com/google/ml-metadata/commit/aa492c474f8b3a54dbb6461ddb7b35c7e69b4a47) not being used to test on the tfx-ci-test in GitHub (guessing internal tests should pass as it uses source files)

You mean mlmd 1.5 release is needed?

**Comment by casassg** [comment]
Yeah. Or some way to run the tests from source (which is not how GitHub Actions does it atm). Note the error is not related to this change, rather is a commit in main tfx and main mlmd. 

When pulling the change, internal Google CI should be green as it runs from source (which also explains why this commit is in the master branch of TFX).

**Comment by 1025KB** [comment]
@jay90099, do you know if we can create CLs when 'merging is blocked'

**Comment by jiyongjung0** [comment]
Hi! I [updated our ci to use nightly packages](https://github.com/tensorflow/tfx/commit/547e7c421c76b74eb87ff743c2b98129274a8d8d) to reflect changes in dependent packages more quickly. But then other tests are failing with incompatible TF version.(However this PR might pass if you didn't touch the related codes.) We are upgrading the base TF version to 2.7, but there are some conflicts in the process. I think that we can proceed the PR if your manual test passed.

**Comment by casassg** [comment]
After https://github.com/tensorflow/tfx/commit/46f424bbbd35ee9218273db19443ec91f06ce2fb seems to be green at least locally. Waiting for this to run here as well.

**Comment by gbaned** [comment]
@casassg Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
@zhitaoli Can you please review this PR ? Thanks!

**Comment by casassg** [comment]
re-up

**Review by jiyongjung0 — APPROVED** [review]
LGTM

File: tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py [hunk 1]
```diff
@@ -58,18 +59,19 @@ def process(
...
-) -> input_pb2.ExampleListWithContext:
+) -> bytes:
   """Convert context feature and examples to ELWC."""
   context_feature, examples = context_feature_and_examples
   context_feature_proto = tf.train.Example()
   context_feature_proto.ParseFromString(context_feature)
-  return input_pb2.ExampleListWithContext(
+  elwc_pb2 = input_pb2.ExampleListWithContext(
       context=context_feature_proto, examples=examples)
+  return elwc_pb2.SerializeToString()
```
Comment (1025KB) [inline tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py:69]: can we add deterministic=True for a deterministic partition?

File: tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py [hunk 2]
```diff
@@ -89,13 +91,21 @@ def _BigQueryToElwc(pipeline: beam.Pipeline, exec_properties: Dict[str, Any],
...
     RuntimeError: Context features must be included in the queried result.
   """
+  # Try to parse the GCP project ID from the beam pipeline options.
+  beam_pipeline_args = exec_properties['_beam_pipeline_args']
+  pipeline_options = beam.options.pipeline_options.PipelineOptions(
+      beam_pipeline_args)
+  project = pipeline_options.view_as(
+      beam.options.pipeline_options.GoogleCloudOptions).project
+  if isinstance(project, value_provider.ValueProvider):
+    project = project.get()
```
Comment (1025KB) [inline tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py:101]: Is this duplicated from BigQueryExampleGen? if so, should we call into a common utility for this (as opposed to duplicating the code)?

File: tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py (hunk 1 above)
Comment (casassg) [reply tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py:69]: Won't this also slow down the serialization process here? Worried on performance vs correctness.

File: tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py (hunk 2 above)
Comment (casassg) [reply tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py:101]: It is indeed duplicate. I can try to refactor into its own library separately. Mostly didn't want to create a separate library as part of this.

What would be a good place? tfx/extensions/google_cloud_big_query/component_utils.py?

Comment (1025KB) [reply tfx/extensions/google_cloud_big_query/experimental/elwc_example_gen/component/executor.py:101]: you can use this tfx/extensions/google_cloud_big_query/utils.py

**Comment by edi-bice-by** [comment]
@jiyongjung0 can you please take another look at the tfx-ci-test?

**Comment by casassg** [comment]
Lmk if you need me to rebase

**Review by 1025KB — APPROVED** [review]

**Comment by 1025KB** [comment]
sorry I missed this, LGTM

**Comment by casassg** [comment]
👀

**Comment by 1025KB** [comment]
Hi, Sanjay, this one is ready to merge, but previous auto-generated internal CL is obsolete, how to re enable the merging process?

**Review by jiyongjung0 — APPROVED** [review]

**Pull request merged**